chef/cookbooks/updater/recipes/default.rb

```diff
@@ -45,6 +48,67 @@
       ignore_failure true
     end
 
+    ruby_block "check for updates" do
+      block do
+        case node[:updater][:zypper][:method]
+        when "patch"
+          command = "list-patches"
+        when "update"
+          command = "list-updates"
+        when "dist-upgrade"
+          command = "list-updates"
+        end
+
+        node.run_state["needs_update"] = `zypper -q #{command}|wc -l`.chomp.to_i > 0
+
+        command += '|egrep -q "corosync|pacemaker"'
+        system("zypper #{command}")
+        # exit 0: found, 1 not found
+        node.run_state["found_ha_packages"] = $?.exitstatus ? true : false
+      end
+      not_if { search(:node, "roles:pacemaker-cluster-member").empty? }
+    end
+
+    ["corosync", "pacemaker"].each do |s|
+      service s do
+        action :stop
+        only_if { node.run_state["found_ha_packages"] }
+        not_if { node[:pacemaker][:is_remote] }
+      end
+    end
+
+    # set cluster to maintenance if
+    # HA packages are NOT gonna be updated
+    # And Node is part of a cluster
+    # And there is packages to update
+    execute "crm --wait node maintenance" do
+      action :nothing
+      notifies :create, "ruby_block[set maintenance mode via this chef run]", :immediately
+    end
+
+    ruby_block "set maintenance mode via this chef run" do
+      action :nothing
+      block do
+        set_maintenance_mode_via_this_chef_run
+      end
+    end
+
+    ruby_block "set cluster maintenance" do
+      block do
+        Chef::Log.info("Triggering maintenance mode for this node")
+        true
+      end
+      only_if do
+        ha = node.run_state["found_ha_packages"]
+        is_cluster = !search(:node, "roles:pacemaker-cluster-member").empty?
```

Comment:
wtf, this is wrong. I wanted to check if the node is part of a cluster, this will just search for any nodes that have that role ¬_¬